src/protocol/errors.rs

```diff
@@ -18,19 +19,65 @@ pub enum TradeSimulationErrorKind {
     U256Overflow,
 }
 
-/// Struct representing a trade simulation error.
+/// Struct representing a native simulation error.
 #[derive(Debug)]
-pub struct TradeSimulationError {
+pub struct NativeSimulationError {
     /// The kind of error that occurred.
     pub kind: TradeSimulationErrorKind,
     /// The partial result of the simulation, if any.
     pub partial_result: Option<GetAmountOutResult>,
 }
 
-impl TradeSimulationError {
+impl NativeSimulationError {
     /// Creates a new trade simulation error with the given kind and partial result.
     pub fn new(kind: TradeSimulationErrorKind, partial_result: Option<GetAmountOutResult>) -> Self {
-        TradeSimulationError { kind, partial_result }
+        NativeSimulationError { kind, partial_result }
+    }
+}
+
+use std::fmt;
+
+impl fmt::Display for TradeSimulationErrorKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            TradeSimulationErrorKind::InsufficientData => {
+                write!(f, "Insufficient data to perform the simulation")
+            }
+            TradeSimulationErrorKind::NoLiquidity => {
+                write!(f, "No liquidity in the venue to complete the trade")
+            }
+            TradeSimulationErrorKind::Unknown => {
+                write!(f, "An unknown error occurred during the simulation")
+            }
+            TradeSimulationErrorKind::InsufficientAmount => {
+                write!(f, "Insufficient amount provided for the trade")
+            }
+            TradeSimulationErrorKind::U256Overflow => {
+                write!(f, "Arithmetic operation resulted in a U256 overflow")
+            }
+        }
+    }
+}
```

Comment:
I understand we had to implement Display here because we are using this weird pattern of having an `ErrorKind` instead of just having a single enum for all errors. I much prefer the usage of the `#[error(...)]` macro over this.